code/game/objects/items/rogueitems/necro_relics.dm

```diff
@@ -12,70 +12,76 @@
     grid_width = 32
 
 /obj/item/necro_relics/necro_crystal/attack_self(mob/living/user)
-	..()
+    ..()
+    if(!user) 
+        return FALSE
+    if(world.time - src.last_use_time < src.use_cooldown)
+        to_chat(user, span_warning("The crystal thrums under your touch, but remains inert."))
+        return FALSE
 
-	if(!user) 
-		return FALSE
-	if(world.time - src.last_use_time < src.use_cooldown)
-		to_chat(user, span_warning("The crystal thrums under your touch, but remains inert."))
-		return FALSE
+    // Ask the Necromancer for a task for the skeleton BEFORE the timer
+    var/task = input(user, "What task do you wish to assign your Greater Skeleton?", "Assign Task") as null|text
```

Comment:
this is either best left as a selection from a pre-set list ("War!", "Resources", "Hunting") etc or html sanitized